### PR TITLE
Add if/else automation/script action

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

This PR adds support for the `if` action, providing If-Then-Else in automations and scripts.

image

Core PR: <https://github.com/home-assistant/core/pull/69811>

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/automation/action/types/ha-automation-action-if.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  protected render() {
+    const action = this.action;
+
+    return html`
+      <h3>
+        ${this.hass.localize(
+          "ui.panel.config.automation.editor.actions.type.if.if"
+        )}:
```
Comment (zsarnett) [inline src/panels/config/automation/action/types/ha-automation-action-if.ts:29]:
```suggestion
        )}*:
```

File: src/panels/config/automation/action/types/ha-automation-action-if.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      <ha-automation-condition
+        .conditions=${action.if}
+        .hass=${this.hass}
+        @value-changed=${this._ifChanged}
+      ></ha-automation-condition>
+
+      <h3>
+        ${this.hass.localize(
+          "ui.panel.config.automation.editor.actions.type.if.then"
+        )}:
```
Comment (zsarnett) [inline src/panels/config/automation/action/types/ha-automation-action-if.ts:40]:
```suggestion
        )}*:
```

File: src/panels/config/automation/action/types/ha-automation-action-if.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    fireEvent(this, "value-changed", {
+      value: {
+        ...this.action,
+        else: value,
+      },
+    });
+  }
+
+  static get styles(): CSSResultGroup {
+    return [haStyle];
```
Comment (zsarnett) [inline src/panels/config/automation/action/types/ha-automation-action-if.ts:96]:
```suggestion
    return haStyle;
```

**Review by zsarnett — DISMISSED** [review]

File: src/panels/config/automation/action/types/ha-automation-action-if.ts (hunk 1 above)
Comment (frenck) [reply src/panels/config/automation/action/types/ha-automation-action-if.ts:29]: So, don't mind if I ask, what does this do? It makes it a generator?

Comment (zsarnett) [reply src/panels/config/automation/action/types/ha-automation-action-if.ts:29]: To show it's required

**Comment by frenck** [comment]
Updated the PR to show defaults on the required fields

CleanShot 2022-04-20 at 16 17 43

**Review by zsarnett — APPROVED** [review]

**Pull request merged**